src/R/Editor/Impl/Data/WorkspaceVariableProvider.cs

```diff
@@ -133,9 +137,8 @@ internal sealed class WorkspaceVariableProvider : RSessionChangeWatcher, IVariab
             try {
                 _updating = true;
                 // May be null in tests
-                var session = Workflow.RSession;
-                if (session.IsHostRunning) {
-                    var stackFrames = await session.TracebackAsync();
+                if (Session.IsHostRunning) {
```

Comment:
I think there may also be a problem if session gets disposed between async calls below. May be worth catching disposed exception too.

Reply:
AFAIK `RHost` doesn't throw `ObjectDisposedException`. When connection is terminated, it throws `RHostDisconnectedException`, which is derived from `OperationCanceledException` and so it will be swallowed by `DoNotWait`.